src/vconsole/vconsole-setup.c

```diff
@@ -43,64 +45,85 @@
 #include "util.h"
 #include "virt.h"
 
+#define MAX_CONSOLES 63
+
 static bool is_vconsole(int fd) {
         unsigned char data[1];
 
         data[0] = TIOCL_GETFGCONSOLE;
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
-static int disable_utf8(int fd) {
-        int r = 0, k;
-
-        if (ioctl(fd, KDSKBMODE, K_XLATE) < 0)
-                r = -errno;
+static bool is_settable(int n, int *fd) {
+        char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
+        int fdt, r, curr_mode;
+
+        /* skip non-allocated ttys */
+        xsprintf(vcname, "/dev/vcs%i", n);
+        if (access(vcname, F_OK) < 0)
+                return false;
+
+        /* try to open terminal */
+        xsprintf(vcname, "/dev/tty%i", n);
+        fdt = open_terminal(vcname, O_RDWR|O_CLOEXEC);
+        if (fdt < 0)
+                return false;
+
+        r = ioctl(fdt, KDGKBMODE, &curr_mode);
+        /*
+         * Make sure we only adjust consoles in K_XLATE or K_UNICODE mode.
+         * Oterwise we would (likely) interfere with X11's processing of the
+         * key events.
+         *
+         * http://lists.freedesktop.org/archives/systemd-devel/2013-February/008573.html
+         */
+        if (r || (curr_mode != K_XLATE && curr_mode != K_UNICODE)) {
```

Comment:
We prefer to an explicit comparison to zero for ints, so please use something like this instead:
```
  if (r != 0 || !IN_SET(curr_mode, K_XLATE, K_UNICODE)) {
```